### PR TITLE
Profile options

### PR DESCRIPTION
**Pull request by paulstapor**

### CODE REVIEW

File: pypesto/profile/profile.py [hunk 1]
```diff
@@ -63,7 +66,12 @@ def parameter_profile(
     # Handling defaults
     # profiling indices
     if profile_index is None:
-        profile_index = np.ones(problem.dim_full)
+        list(range(problem.dim_full))
```
Comment (yannikschaelte) [inline pypesto/profile/profile.py:69]: exclude fixed indices.

File: pypesto/profile/profile.py [hunk 2]
```diff
@@ -39,9 +40,11 @@ def parameter_profile(
     optimizer:
         The optimizer to be used along each profile.
     profile_index:
+        List with the profile indices to be computed (by default all of them).
+    profile_index_advanced:
```
Comment (yannikschaelte) [inline pypesto/profile/profile.py:44]: not sure if there is a better name ... ;)

File: pypesto/profile/profile.py [hunk 3]
```diff
@@ -63,7 +66,12 @@ def parameter_profile(
     # Handling defaults
     # profiling indices
     if profile_index is None:
-        profile_index = np.ones(problem.dim_full)
+        list(range(problem.dim_full))
+
+    # check if more detailed options were passed
+    if profile_index_advanced is None:
+        profile_index_advanced = np.array([0] * problem.dim_full)
+        profile_index_advanced[profile_index] = 1
```
Comment (yannikschaelte) [inline pypesto/profile/profile.py:74]: profile_index_advances is currently not used yet, so maybe exclude for the moment? line 94/102 does not seem to fit to the new form of profile_index yet. nbs would need to be updated too.

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/profile/profile.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5762992&bid=18864629)

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/profile/profile.py  1
         

Complexity decreasing per file
==============================
+ pypesto/profile/approximate.py  -1
+ pypesto/profile/util.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5762992&bid=18864629)

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/profile/profile.py  1
         

Complexity decreasing per file
==============================
+ pypesto/profile/approximate.py  -1
+ pypesto/profile/util.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5762992&bid=18864629)

File: pypesto/profile/profile.py [hunk 4]
```diff
@@ -18,7 +18,8 @@ def parameter_profile(
         problem: Problem,
         result: Result,
         optimizer: Optimizer,
-        profile_index: np.ndarray = None,
+        profile_index: Union[np.ndarray, list] = None,
+        profile_index_advanced: Union[np.ndarray, list] = None,
```
Comment (yannikschaelte) [inline pypesto/profile/profile.py:22]: this is not used and not documented

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/profile/profile.py  1
         

Complexity decreasing per file
==============================
+ pypesto/profile/approximate.py  -1
+ pypesto/profile/util.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5762992&bid=18864629)

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/profile/profile.py  1
         

Complexity decreasing per file
==============================
+ pypesto/profile/approximate.py  -1
+ pypesto/profile/util.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5762992&bid=18864629)

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/profile/profile.py  1
         

Complexity decreasing per file
==============================
+ pypesto/profile/approximate.py  -1
+ pypesto/profile/util.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5762992&bid=18864629)

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/profile/profile.py  1
         

Complexity decreasing per file
==============================
+ pypesto/profile/approximate.py  -1
+ pypesto/profile/util.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5762992&bid=18864629)

File: pypesto/profile/profile.py [hunk 5]
```diff
@@ -192,8 +190,16 @@ def walk_along_profile(
         # run optimization
         # IMPORTANT: This optimization will need a proper exception
         # handling (coming soon)
-        optimizer_result = optimizer.minimize(problem, startpoint, '0',
-                                              allow_failed_starts=False)
+        if startpoint.size > 0:
```
[the remark this answers is not in the document]
Comment (paulstapor) [reply pypesto/profile/profile.py:193]: Good idea. I'm anyhow fan of "pimp my coverage". :)

[the remark this answers is not in the document]
Comment (paulstapor) [reply pypesto/profile/profile.py:193]: Found indeed a bug here

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/profile/profile.py  1
         

Complexity decreasing per file
==============================
+ pypesto/profile/approximate.py  -1
+ pypesto/profile/util.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5762992&bid=18864629)

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/profile/profile.py  1
         

Complexity decreasing per file
==============================
+ pypesto/profile/approximate.py  -1
+ pypesto/profile/util.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5762992&bid=18864629)

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/profile/profile.py  1
         

Complexity decreasing per file
==============================
+ pypesto/profile/approximate.py  -1
+ pypesto/profile/util.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5762992&bid=18864629)

**Pull request merged**